### PR TITLE
Automatically port Reflection.PortableExecutable.PdbChecksumDebugDirectoryData

### PR DESCRIPTION
**Pull request by carlossanlop**

### CODE REVIEW

File: xml/System.Reflection.PortableExecutable/PdbChecksumDebugDirectoryData.xml [hunk 1]
```diff
@@ -41,7 +41,7 @@
         <ReturnType>System.String</ReturnType>
       </ReturnValue>
       <Docs>
-        <summary>To be added.</summary>
+        <summary>Checksum algorithm name.</summary>
```
Comment (rpetrusha) [inline xml/System.Reflection.PortableExecutable/PdbChecksumDebugDirectoryData.xml]:
```suggestion
        <summary>Gets the checksum algorithm name.</summary>
```
But is this a "checksum algorithm name" or a "checksum based on the algorithm name"?

File: xml/System.Reflection.PortableExecutable/PdbChecksumDebugDirectoryData.xml [hunk 2]
```diff
@@ -41,7 +41,7 @@
...
         <value>To be added.</value>
```
Comment (rpetrusha) [inline xml/System.Reflection.PortableExecutable/PdbChecksumDebugDirectoryData.xml]:
```suggestion
        <value>The checksum algorithm name.</value>
```

File: xml/System.Reflection.PortableExecutable/PdbChecksumDebugDirectoryData.xml [hunk 3]
```diff
@@ -63,7 +63,7 @@
         <ReturnType>System.Collections.Immutable.ImmutableArray&lt;System.Byte&gt;</ReturnType>
       </ReturnValue>
       <Docs>
-        <summary>To be added.</summary>
+        <summary>GUID (Globally Unique Identifier) of the associated PDB.</summary>
```
Comment (rpetrusha) [inline xml/System.Reflection.PortableExecutable/PdbChecksumDebugDirectoryData.xml]:
```suggestion
        <summary>Gets the GUID (Globally Unique Identifier) of the associated PDB.</summary>
```

File: xml/System.Reflection.PortableExecutable/PdbChecksumDebugDirectoryData.xml [hunk 4]
```diff
@@ -63,7 +63,7 @@
...
         <value>To be added.</value>
```
Comment (rpetrusha) [inline xml/System.Reflection.PortableExecutable/PdbChecksumDebugDirectoryData.xml]:
```suggestion
        <value>The GUID of the associated PDB.</value>
```

**Comment by carlossanlop** [comment]
@tmat would you mind helping us? Since you introduced this type, we need:

- A type summary. There isn't one in the source. I tried coming up with a description based on the text in the PR and the proposal, but I would prefer it the type creator gives us the most accurate description 😃 
- Address @rpetrusha 's [question ](https://github.com/dotnet/dotnet-api-docs/pull/3194#discussion_r328388347) about `HashAlgorithmName`: _"is this a "checksum algorithm name" or a "checksum based on the algorithm name"?"_

Here's the source: [src/System.Reflection.Metadata/src/System/Reflection/PortableExecutable/DebugDirectory/PdbChecksumDebugDirectoryData.cs](https://github.com/dotnet/corefx/blob/a10890f4ffe0fadf090c922578ba0e606ebdd16c/src/System.Reflection.Metadata/src/System/Reflection/PortableExecutable/DebugDirectory/PdbChecksumDebugDirectoryData.cs)

Here's the PR that introduced: https://github.com/dotnet/corefx/pull/26976

Here's the proposal (the text is the same as in the PR): https://github.com/dotnet/corefx/issues/26935

**Comment by tmat** [comment]
The type represents a _PDB Checksum_ debug directory entry as specified here: https://github.com/dotnet/corefx/blob/master/src/System.Reflection.Metadata/specs/PE-COFF.md#pdb-checksum-debug-directory-entry-type-19.

[This](https://github.com/dotnet/corefx/blob/a10890f4ffe0fadf090c922578ba0e606ebdd16c/src/System.Reflection.Metadata/src/System/Reflection/PortableExecutable/DebugDirectory/PdbChecksumDebugDirectoryData.cs#L17-L19) comment is incorrect. Should be "Checksum of the PDB content.".

File: xml/System.Reflection.PortableExecutable/PdbChecksumDebugDirectoryData.xml (hunk 1 above)
Comment (tmat) [reply xml/System.Reflection.PortableExecutable/PdbChecksumDebugDirectoryData.xml]: The name of crypto hash algorithm used to calculate the checksum. E.g. "SHA256".

**Comment by carlossanlop** [comment]
Thank you, @tmat for your quick help. I updated the PR with your suggestions. Let @rpetrusha and I know if you agree with the changes I made so we can merge this.

File: xml/System.Reflection.PortableExecutable/PdbChecksumDebugDirectoryData.xml (hunk 3 above)
Comment (carlossanlop) [reply xml/System.Reflection.PortableExecutable/PdbChecksumDebugDirectoryData.xml]: This description was wrong as mentioned in the other comment. I updated it.

File: xml/System.Reflection.PortableExecutable/PdbChecksumDebugDirectoryData.xml (hunk 4 above)
Comment (carlossanlop) [reply xml/System.Reflection.PortableExecutable/PdbChecksumDebugDirectoryData.xml]: This description was wrong as mentioned in the other comment. I updated it.

File: xml/System.Reflection.PortableExecutable/PdbChecksumDebugDirectoryData.xml (hunk 3 above)
Comment (carlossanlop) [reply xml/System.Reflection.PortableExecutable/PdbChecksumDebugDirectoryData.xml]: This description was wrong as mentioned in the other comment. I updated it.

**Comment by rpetrusha** [comment]
Is this ready to merge, @carlossanlop?

**Comment by carlossanlop** [comment]
Yes, it's ready to merge, @rpetrusha.

**Pull request merged**